inspirehep/modules/converttohal/tei.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/python
+
+from jinja2 import Environment, PackageLoader
+import json
+
+TEMPLATE = 'default.xml'
+#from inspirehep.utils.record_getter import get_es_record
+from inspirehep.modules.records.json_ref_loader import replace_refs
+#from inspirehep.modules.authors.utils import scan_author_string_for_phrases
+from nameparser import HumanName
+import sys
+
+
+def tei_response(record):
+    
+    data = record
+    env = Environment(loader=PackageLoader('inspirehep.modules.converttohal',
+                                           'templates'), trim_blocks=True, lstrip_blocks=True)
+    template = env.get_template(TEMPLATE)
+    #import ipdb; ipdb.set_trace()
+
+    authors = data.get('authors', [])
+    for author in authors:
+        if 'full_name' in author and author['full_name']:
+            parsed = HumanName(author['full_name'])
+            author['parsed_name'] = parsed
+
+    titles = data.get('titles', [])
+
+    doi = data['dois'][0]['value'] if 'dois' in data else ""
+
+    if 'publication_info' in data:
```

Comment:
Have a look at EAFP (Easier to ask forgiveness than permission): https://docs.python.org/2/glossary.html#term-eafp